sdk/ai/azure-ai-projects/tests/agents/test_agents_tools.py

```diff
@@ -0,0 +1,299 @@
+# # ------------------------------------
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+# ------------------------------------
+import os
+import json
+import time
+import functools
+import datetime
+import logging
+import sys
+
+from azure.ai.projects import AIProjectClient
+from azure.ai.projects.models import FunctionTool, CodeInterpreterTool, FileSearchTool, ToolSet, VectorStore
+from azure.core.pipeline.transport import RequestsTransport
+from devtools_testutils import AzureRecordedTestCase, EnvironmentVariableLoader, recorded_by_proxy
+from azure.core.exceptions import AzureError, ServiceRequestError, HttpResponseError
+from azure.identity import DefaultAzureCredential
+
+# TODO clean this up / get rid of anything not in use
+
+'''
+issues I've noticed with the code: 
+    delete_thread(thread.id) fails
+    cancel_thread(thread.id) expires/times out occasionally
```

Comment:
Maybe good to have issue about this also, is there a reason information in cancellation?